gherkin-tests/src/main.rs

```diff
@@ -0,0 +1,107 @@
+use std::{cmp::Ordering, sync::Arc};
+
+use cucumber::{given, then, when, WorldInit};
+use many_client::client::ledger::TokenAmount;
+use num_bigint::BigUint;
+use opts::{read_spec_config, CmdOpts};
+use world::{IdentityName, World};
+
+mod cose;
+mod opts;
+mod world;
+
+#[given(expr = "an identity {identity}")]
+fn setup_identity(world: &mut World, id: IdentityName) {
+    world.insert_identity(id);
+}
+
+#[given(expr = "a symbol {word}")]
+fn setup_symbol(world: &mut World, symbol: String) {
+    assert!(world.symbols().contains_key(&symbol));
+}
+
+#[given(expr = "{identity} has {int} {word}")]
+async fn id_has_x_symbols(world: &mut World, id: IdentityName, amount: BigUint, symbol: String) {
+    let amount: TokenAmount = amount.into();
+    let faucet = world.spec_config().faucet_identity.clone();
+    let identity = world.identity(&id).unwrap().clone();
+    let symbol = *world.symbol(&symbol).unwrap();
+    let current_balance = world.balance(identity.identity, symbol).await;
+
+    let faucet_balance = world.balance(faucet.identity, symbol).await;
+    assert_ne!(faucet_balance, TokenAmount::zero());
+
+    match amount.cmp(&current_balance) {
+        Ordering::Greater => world
+            .ledger_client()
+            .send(
+                faucet,
+                identity.identity,
+                amount.clone() - current_balance,
+                symbol,
+            )
+            .await
+            .expect("Should have sent"),
+        Ordering::Less => world
+            .ledger_client()
+            .send(
+                identity.clone(),
+                faucet.identity,
+                current_balance - amount.clone(),
+                symbol,
+            )
+            .await
+            .expect("Should have sent"),
```

Comment:
Is there a use case for this?

Reply:
1. Tests were we do some operations over an account, and then want to reset its money for a bunch of more steps
2. Completeness